### DISCUSSION TITLE
[功能咨询]双语字幕功能是否是同一时刻翻译所有外语字幕

### DISCUSSION THREAD

**Discussion by zhatlas** [immersive-translate/immersive-translate#1266] [Help]
### 概述 | Summary

比如在观看 YouTube 两小时英语视频，有英文字幕，那我进入这个视频观看的时候，沉浸式翻译插件是否是在同一时刻调用 API 翻译了所有字幕？
如果是，那就会存在一个问题，我可能只是想看一下两小时视频说了什么，看了很短时间我就不感兴趣了，但是沉浸式翻译浪费了大量 token 去翻译所有字幕。
如果不是，那沉浸式翻译的字幕翻译策略是什么？

**Comment by zhatlas** [comment]
邀请回答一下🙏 @theowenyoung

**Reply by theowenyoung** [reply]
默认情况下，Youtube 视频字幕使用的是官方网站自带的机器中文字幕，所以不会消耗任何流量。

如果你主动开启了优先使用沉浸式翻译来翻译的话，那么对于 Youtube 来说，是的，沉浸式翻译在视频一开始就会翻译整个视频的字幕。
